util/eventbus/subscribe.go

```diff
@@ -4,46 +4,59 @@
 package eventbus
 
 import (
+	"context"
 	"fmt"
 	"reflect"
 	"sync"
 )
 
-type dispatchFn func(vals *queue, stop goroutineShutdownWorker, acceptCh func() chan any) bool
+// subscriber is a uniformly typed wrapper around Subscriber[T], so
+// that debugging facilities can look at active subscribers.
+type subscriber interface {
```

Comment:
Since this is meant to be for the debugger, does it need to be an interface, or could it be a struct with fields on it? (Not that it strictly matters, but if the type is the only variable, then I think a `reflect.Type` covers us there)

Reply:
Hmm, I might be doable as a struct with fields, but we still need the exported Subscriber[T] to present to users. An interface allows that family of types to all implement the one thing, rather than two values floating around, so to speak.

I dunno, no strong preference. Right now I'm mimicking the interface type for publishers, so if we change this we should change both.

Reply:
This seems fine, and your explanation makes sense to me.